test/e2e/page-objects/pages/dialog/add-tokens.ts

```diff
@@ -34,12 +34,14 @@ class AddTokensModal {
    * @param expectedTokenCount - The expected count of suggested tokens.
    */
   async checkSuggestedTokensCount(expectedTokenCount: number) {
-    const multipleSuggestedTokens = await this.driver.findElements(
-      this.tokenListItem,
-    );
-
     // Confirm the expected number of tokens are present as suggested token list
-    assert.equal(multipleSuggestedTokens.length, expectedTokenCount);
+    await this.driver.wait(async () => {
+      const multipleSuggestedTokens = await this.driver.findElements(
+        this.tokenListItem,
+      );
+
+      return multipleSuggestedTokens.length === expectedTokenCount;
+    }, 10000);
```

Comment:
sometimes the wallet takes time loading the token entries -- with the previous method there could be race conditions where there we find that element, and has not yet loaded all the tokens.
findElements+assert is a pattern that can lead to race conditions. 
Waiting for our expected condition adds robustness to this assertion